File: src/main/java/org/dedda/bratwurst/parse/StringParser.java
package org.dedda.bratwurst.parse;

import org.dedda.bratwurst.lang.BWString;

/**
 * Created by dedda on 12/10/15.
 *
 * @author dedda
 */
public class StringParser extends ExpressionParser {

    @Override
    public BWString parse(String data, int lineNumber) {
        if (data.matches(Patterns.BW_STRING)) {
            String message = data.substring(Emoji.POODLE.length(), data.length() - Emoji.POOP.length());
            return new BWString(message);
        }
        return null;
    }

}

File: src/main/java/org/dedda/bratwurst/parse/StringParser.kt
package org.dedda.bratwurst.parse

import org.dedda.bratwurst.lang.BWString

/**
 * Created by dedda on 12/10/15.
 *
 * @author dedda
 */
class StringParser : ExpressionParser() {
    override fun parse(data: String, lineNumber: Int): BWString? {
        if (data.matches(PATTERN)) {
            val message = data.substring(1, data.length - 1)
            return BWString(message)
        }
        return null
    }

    companion object {
        private val PATTERN = Regex(Patterns.BW_STRING)
    }
}